service/host.go

```diff
@@ -149,6 +149,13 @@ func (s *host) shutdown(err error, reason string, exitCode *int) (bool, error) {
 		s.scope.Close()
 	}
 
+	// Flush tracing buffers
+	if s.tracerCloser != nil {
+		if err = s.tracerCloser.Close(); err != nil {
+			ulog.Logger().Error("Failure to close tracer", "error", err)
```

Comment:
This is the global logger. This should use `s.Logger()` instead of the global one (there's also a mistake above which is probably why you did this, you don't need to necessarily fix that one here but we should make a note to do it later)